### DISCUSSION TITLE
How to get rid of deprecation warnings in RMQ UI/Landing Page

### DISCUSSION THREAD

**Discussion by HunterTheWalrus** [rabbitmq/rabbitmq-server#11978] [Questions] [answered]
I'm in a bit of a pickle here. We have a release coming up and I was tasked with upgrading to RabbitMQ 3.13.5. We are still using the classic mirrored queues, and I have been told that we will not be moving onto quorum queues. I'm not sure exactly what our plans are since it is going away in 4.0. We also require customers to set up their RMQ the same way that ours are set up. We deliver custom nuget pkgs for them etc. My main issue is this: I cannot figure out a way to get rid of this "Deprecated features are being used" warning at the top of the RabbitMQ landing page/UI. 

![image](https://github.com/user-attachments/assets/0107b417-7897-4cd8-8bdd-3e0b8cc46e21)

I have edited the rabbitmq.conf and added the following lines: deprecated_features.permit.my_deprecated_feature = true   deprecated_features.permit.classic_queue_mirroring = true  

![image](https://github.com/user-attachments/assets/d46240da-b0f0-4bd8-9404-857a3bf0d71d)

 I thought that those lines would be the fix, but the warning remains and I really really need to get rid of it. Is that even a possibility? 

Edit: This will be a temporary thing since we have a release coming up, next release we will have moved over to quorum queues. We just need to clear that warning since that UI is customer facing.

**Comment by lukebakken** [comment, marked as answer]
```
lbakken@PROKOFIEV ~/development/rabbitmq/rabbitmq-server_v3.13.x (v3.13.x =)
$ git branch
* v3.13.x

lbakken@PROKOFIEV ~/development/rabbitmq/rabbitmq-server_v3.13.x (v3.13.x =)
$ git grep 'Deprecated features are being used'
deps/rabbitmq_management/priv/www/js/main.js:438:        l1 += '<span>&#9888;</span> Deprecated features are being used. <a href="https://www.rabbitmq.com/feature-flags.html">[Learn more]</a>'
deps/rabbitmq_management/priv/www/js/tmpl/deprecated-features.ejs:13:        Deprecated features are being used. While using deprecated features, upgrading to future minor or major versions of RabbitMQ may not be possible. <a href="https://www.rabbitmq.com/feature-flags.html">[Learn more]</a>
```

https://github.com/rabbitmq/rabbitmq-server/blob/v3.13.x/deps/rabbitmq_management/priv/www/js/main.js#L416-L449

It looks as though as long as you are using any deprecated features, you will see that warning. Options, off the top of my head:
* Edit that file in your deployments to remove the warning
* Use a client-side hack to do the same as what this code does - https://github.com/rabbitmq/rabbitmq-server/blob/v3.13.x/deps/rabbitmq_management/priv/www/js/main.js#L446-L447

**Reply by HunterTheWalrus** [reply]
I will see if this is something our team is willing to do, test, then mark as answer. Thank you so much for your response.

**Reply by HunterTheWalrus** [reply]
We determined that we do not want to edit the source code. We will just be adding this to our documentation instead. Thanks for your response, marking resolved.
